src/base/bittorrent/session.cpp

```diff
@@ -4513,6 +4514,43 @@ void Session::startUpTorrents()
         if (needStore)
              m_resumeDataStorage->store(torrentID, resumeData);
 
+        const QString category = resumeData.category;
+        bool isCategoryRecovered = recoveredCategories.contains(category);
+        if (!category.isEmpty() && (isCategoryRecovered || !m_categories.contains(category)))
+        {
+            if (!isCategoryRecovered)
+            {
+                if (addCategory(category))
+                {
+                    recoveredCategories.insert(category);
+                    isCategoryRecovered = true;
+                    LogMsg(tr("Inconsistent data is detected. Category '%2' is assigned to some torrent(s) but doesn't exist. It will be recovered with default settings.")
```

Comment:
>Category '%2' is assigned to some torrent(s) but doesn't exist.

To make it more specific: `Category '%2' is assigned to some torrent(s) but doesn't exist in configuration file`. If "configuration file" is incorrect, please substitute it.